node/src/run_modes_factories.rs

```diff
@@ -0,0 +1,447 @@
+    ) {
+        let daemon_clustered_params = test_clustered_params();
+        let subject = DaemonInitializerFactoryReal::new(
+            Box::new(NodeConfiguratorInitializationReal),
+            daemon_clustered_params,
+        );
+        let args =
+            convert_str_vec_slice_into_vec_of_strings(&["program", "--wooooooo", "--fooooooo"]);
+        let mut stream_holder = FakeStreamHolder::default();
+
+        let result = subject.make(&args, &mut stream_holder.streams());
+
+        let mut config_error = result.err().unwrap();
+        let actual_error = config_error.param_errors.remove(0);
+        assert!(config_error.is_empty());
+        assert_eq!(actual_error.parameter.as_str(), "<unknown>");
+        assert!(
+            actual_error
+                .reason
+                .contains("Unfamiliar message: error: Found argument '--wooooooo'"),
+            "{}",
+            actual_error.reason
+        );
+    }
+
+    #[test]
+    fn make_for_daemon_initializer_factory_produces_an_error_when_trying_to_pass_through_configure()
+    {
+        let configure_params_arc = Arc::new(Mutex::new(vec![]));
+        let daemon_clustered_params = test_clustered_params();
+        let subject = DaemonInitializerFactoryReal::new(
+            Box::new(
+                NodeConfiguratorInitializationMock::default()
+                    .configure_params(&configure_params_arc)
+                    .configure_result(Err(ConfiguratorError::required("parameter", "too bad")))
+                    .demanded_values_from_multi_config(vec!["ui-port".to_string()]),
+            ),
+            daemon_clustered_params,
+        );
+        let args = convert_str_vec_slice_into_vec_of_strings(&["program", "--initialization"]);
+        let mut stream_holder = FakeStreamHolder::default();
+
+        let result = subject.make(&args, &mut stream_holder.streams());
+
+        let mut config_error = result.err().unwrap();
+        let actual_error = config_error.param_errors.remove(0);
+        assert!(config_error.is_empty());
+        assert_eq!(actual_error.parameter.as_str(), "parameter");
+        assert_eq!(actual_error.reason.as_str(), "too bad");
+        let mut configure_params = configure_params_arc.lock().unwrap();
+        assert_eq!(
+            *configure_params.remove(0).arg_matches_requested_entries,
+            vec!["5333".to_string()]
+        )
+    }
+}
+
+#[cfg(test)]
+pub mod mocks {
+    use crate::run_modes_factories::{
+        DaemonInitializer, DaemonInitializerFactory, DumpConfigRunner, DumpConfigRunnerFactory,
+        RunModeResult, ServerInitializer, ServerInitializerFactory,
+    };
+    use futures::{Async, Future};
+    use masq_lib::command::StdStreams;
+    use masq_lib::shared_schema::ConfiguratorError;
+    use std::cell::RefCell;
+    use std::sync::{Arc, Mutex};
+
+    pub struct DumpConfigRunnerFactoryMock {
```

Comment:
These are pretty easy to use, but they'd be easier to use quickly if they were in standard form, with `.make_params` and `.make_results`.